core/src/main/java/com/airbnb/aerosolve/core/util/Util.java

```diff
@@ -379,6 +381,22 @@ public int compare(DebugScoreDiffRecord e1, DebugScoreDiffRecord e2) {
     return debugDiffRecord;
   }
 
+  public static <T> Set<T> getIntersection(Set<T> a, Set<T> b) {
+    if (a == null) {
+      return b;
```

Comment:
This logic is not consistent with the name `getIntersection`, since by definition, this should return null if one of the set is empty. We may let this `getIntersection` do what it should do and move this logic to the cross transform.